packages/ember/lib/index.js

```diff
@@ -181,8 +181,10 @@ Object.defineProperty(Ember, 'LOG_VERSION', {
 });
 
 Object.defineProperty(Ember, 'MODEL_FACTORY_INJECTIONS', {
-  get()      { return ENV.MODEL_FACTORY_INJECTIONS;    },
-  set(value) { ENV.MODEL_FACTORY_INJECTIONS = !!value;  },
+  get()      { return false; },
+  set(value) {
+    Ember.logger.warn('As of 2.13.x MODEL_FACTORY_INJECTIONS has no power, feel free to remove it from app/app.js');
```

Comment:
Assertion sounds good but I don't think we can strip the flag as users who ignore the deprecation and set the flag to true will have a bad time.